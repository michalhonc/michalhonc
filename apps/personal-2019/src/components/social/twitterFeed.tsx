import * as React from "react";
import axios from "axios";

import { SocialWrapper } from "./socialWrapper";
import { State, useFetch } from "../../hooks/useFetch";
import Twitter from "../../images/twitter.svg";
import Like from "../../images/like.svg";
import Comment from "../../images/comment.svg";

interface ITwitter {
  createdAt: string;
  text: string;
  small: string | null;
  thumb: string | null;
  id: number;
  url: string;
}

export const TwitterFeed = () => {
  const { posts, state } = useFetch<ITwitter>("getTwitterPosts");

  if (state === State.Err) return null;

  return (
    <SocialWrapper
      Icon={Twitter}
      profile="@michalhonc"
      profileUrl="https://twitter.com/Michalhonc"
      className="TwitterFeed"
    >
      {state === State.Loading && <span>loading</span>}
      {posts.map((post: ITwitter) => (
        <a href={post.url} key={post.id}>
          <div
            className="TwitterFeed-tweet"
            dangerouslySetInnerHTML={{ __html: underlineHashtags(post.text) }}
          />
          <div className="TwitterFeed-imageWrapper">
            {post.small && (
              <img
                loading="lazy"
                width="120"
                height="80"
                src={post.small}
                alt="@michalhonc Twitter profile"
              />
            )}
          </div>
        </a>
      ))}
    </SocialWrapper>
  );
};

function underlineHashtags(text: string) {
  return text
    .split(" ")
    .map((str) =>
      str.startsWith("#") || str.startsWith("@")
        ? `<span class="underline">${str}</span>`
        : str
    )
    .join(" ");
}
