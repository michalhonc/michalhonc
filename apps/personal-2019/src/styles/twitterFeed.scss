.TwitterFeed {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;

  & a {
    display: flex;
    align-items: center;

    @include respond(phone) {
      &:not(:last-of-type) {
        padding-bottom: 1.5rem;
      }
    }

    &:hover {
      & .TwitterFeed-tweet {
        text-decoration: underline;
      }

      & img {
        transform: scale(1.1) rotate(1deg);
      }
    }
  }

  &-imageWrapper {
     display: flex;
     overflow: hidden;
     flex-shrink: 0;
  }

  & img {
    transition: all 0.2s;
    margin-left: 0.2rem;

    @include respond(phone) {
      display: none;
    }
  }

  &-tweet {
    color: var(--c-beta);
    font-size: 1.5rem;
    line-height: 1.8rem;

  }
}
