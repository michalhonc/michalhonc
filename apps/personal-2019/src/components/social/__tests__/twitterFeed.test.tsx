import * as React from "react";
import renderer from "react-test-renderer";
import { TwitterFeed } from "../twitterFeed";

describe("twitterFeed", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TwitterFeed />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
