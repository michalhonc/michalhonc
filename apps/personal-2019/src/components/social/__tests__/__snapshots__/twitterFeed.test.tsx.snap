// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`twitterFeed renders correctly 1`] = `
<div
  className="Social-wrapper"
>
  <div
    className="Social-icon"
  >
    <test-file-stub />
  </div>
  <a
    className="underline Social-profile"
    href="https://twitter.com/Michalhonc"
  >
    @michalhonc
  </a>
  <div
    className="Social-content TwitterFeed"
  />
</div>
`;
